src/kohlrahbi/unfoldedahb/unfoldedahbtable.py

```diff
@@ -368,22 +369,33 @@ def convert_to_flat_ahb(self) -> FlatAnwendungshandbuch:
             )
             raise
 
-    def dump_flatahb_json(self, output_directory_path: Path) -> None:
+    def get_flatahb_json_file_path(self, output_directory_path: Path) -> Path:
         """
-        Converts the unfolded AHB to a flat AHB and writes it to a json file.
-        The file will be stored in the directory:
-            'output_directory_path/<edifact_format>/flatahb/<pruefidentifikator>.json'
+        returns the filepath to where the flat ahb json will be dumped when using dump_flatahb_json()
+        raises a value error when the pruefidentifikator is not a valid one
         """
         edifact_format = get_format_of_pruefidentifikator(self.meta_data.pruefidentifikator)
         if edifact_format is None:
             logger.warning("'%s' is not a pruefidentifikator", self.meta_data.pruefidentifikator)
-            return
+            raise ValueError(f"'{self.meta_data.pruefidentifikator}' is not a pruefidentifikator")
 
         flatahb_output_directory_path = output_directory_path / str(edifact_format) / "flatahb"
-        flatahb_output_directory_path.mkdir(parents=True, exist_ok=True)
-        flat_ahb = self.convert_to_flat_ahb()
-
         file_path = flatahb_output_directory_path / f"{self.meta_data.pruefidentifikator}.json"
+        return file_path
+
+    def dump_flatahb_json(self, output_directory_path: Path) -> None:
+        """
+        Converts the unfolded AHB to a flat AHB and writes it to a json file.
+        The file will be stored in the directory:
+            'output_directory_path/<edifact_format>/flatahb/<pruefidentifikator>.json'
+        """
+        try:
+            file_path = self.get_flatahb_json_file_path(output_directory_path)
+        except ValueError:
+            return
```

Comment:
können wir hier noch was loggen damit es nicht lautlos untergeht?

Reply:
wird schon in der funktion geloggt, die den value error auslöst.
bin mit dem design auch nicht 100% zufrieden, weil wir jetzt an 3 stellen die slebe log message haben, aber ich denke es ist ok. der erste auslöse wird dafür sorgen, dass wir hier in den except block laufen, sodass am ende im log nur 1 mal die message auftaucht.